src/SignalR/clients/csharp/Client.Core/src/HubConnectionBuilderExtensions.cs

```diff
@@ -23,5 +24,49 @@ public static IHubConnectionBuilder ConfigureLogging(this IHubConnectionBuilder
             hubConnectionBuilder.Services.AddLogging(configureLogging);
             return hubConnectionBuilder;
         }
+
+        /// <summary>
+        /// Configures the <see cref="HubConnection"/> to automatically attempt to reconnect if the connection is lost.
+        /// The client will wait the default 0, 2, 10 and 30 seconds respectively before trying up to four reconnect attempts.
+        /// </summary>
+        /// <param name="hubConnectionBuilder">The <see cref="IHubConnectionBuilder" /> to configure.</param>
+        /// <returns>The same instance of the <see cref="IHubConnectionBuilder"/> for chaining.</returns>
+        public static IHubConnectionBuilder WithAutomaticReconnect(this IHubConnectionBuilder hubConnectionBuilder)
+        {
+            // REVIEW: Should we throw if this would override another IRetryPolicy?
```

Comment:
> Should we throw if this would override another IRetryPolicy?

I think this sounds reasonable

Reply:
Considering this is effectively adding a service, I feel it would be more consistent with the rest of the framework to have the last reconnect configuration win.